### PR TITLE
turn off CONFIG_ESP_SYSTEM_HW_STACK_GUARD for all relevant Espressif builds

### LINKED ISSUE #9735: crash when using HTTPServer
@dhalbert Same issue

`Adafruit CircuitPython 9.2.0-beta.1 on 2024-10-11; Maker Go ESP32C3 Supermini with ESP32-C3`

As far as I can tell, `server.poll()` might be the cause.

Currently using the following code.py that still crashes:
```
import wifi
import socketpool
from adafruit_httpserver import Server

pool = socketpool.SocketPool(wifi.radio)
server = Server(pool, debug=False)

while True:
   try:
      server.poll() 

   except Exception as e:
      print(str(e))
```

_Originally posted by @c--- in https://github.com/adafruit/circuitpython/issues/9173#issuecomment-2424948178_

### LINKED ISSUE #9749: Hard fault crash when using HTTPServer: ESP32-C3
Running this test program, with a `settings.toml` containing  valid values for `CIRCUITPY_WIFI_SSID` and `CIRCUITPY_WIFI_PASSWORD`, will cause a hard fault with a few minutes.

It takes around 1-5 minutes to restart in safe mode, though I have seen it run longer. No interaction with the HTTP server is needed. I simply restart code.py and wait.
```py
import wifi
import socketpool
from adafruit_httpserver import Server
import time

time.sleep(5)
print("IP addr", wifi.radio.ipv4_address)

pool = socketpool.SocketPool(wifi.radio)
server = Server(pool, debug=False)

# port 80 will get EADDRINUSE due to web workflow
# Either IP adddress will cause a crash eventually.
server.start("0.0.0.0", 8000)
#server.start(str(wifi.radio.ipv4_address), 8000)

while True:
   try:
      server.poll()
   except Exception as e:
      print(type(e))
```
Either `server.start()` can cause a crash: the wildcard IP address or the actual IP address don't matter. 

I have reproduced this only on a QT Py ESP32-C3. I cannot reproduce on a QT ESP32-S3, with no PSRAM.

[replaces #9735]

### PR DESCRIPTION
**Pull request by dhalbert**
Fixes #9749.

The MicroPython folks discovered this ESP-IDF bug: https://github.com/espressif/esp-idf/issues/14456, which can be worked around by doing `CONFIG_ESP_SYSTEM_HW_STACK_GUARD=n` on both ESP32-C3 and ESP32-C6. See
https://github.com/espressif/esp-idf/issues/14456
https://github.com/micropython/micropython/issues/15667
https://github.com/micropython/micropython/pull/15771
https://github.com/micropython/micropython/pull/15997
I saw this initially and kept in mind. It sounded similar to #9749: a crash after a few minutes on C3 when using wifi.

I tested on a C3. With this fix the test program in #9749 ran for hours before I manually stopped it. Without it, it usually got a hard fault within a few minutes.

### CODE REVIEW

File: ports/espressif/esp-idf-config/sdkconfig-esp32c3.defaults [hunk 1]
```diff
@@ -46,4 +46,7 @@ CONFIG_NEWLIB_NANO_FORMAT=y
 
 # end of Component config
 
+# Workaround for https://github.com/espressif/esp-idf/issues/14456
+CONFIG_ESP_SYSTEM_HW_STACK_GUARD=n
```
Comment (tannewt) [inline ports/espressif/esp-idf-config/sdkconfig-esp32c3.defaults:50]:
```suggestion
# CONFIG_ESP_SYSTEM_HW_STACK_GUARD is not set
```

File: ports/espressif/esp-idf-config/sdkconfig-esp32c6.defaults [hunk 2]
```diff
@@ -51,4 +51,7 @@ CONFIG_ESP_WIFI_STATIC_RX_BUFFER_NUM=4
 
 # end of Component config
 
+# Workaround for https://github.com/espressif/esp-idf/issues/14456
+CONFIG_ESP_SYSTEM_HW_STACK_GUARD=n
```
Comment (tannewt) [inline ports/espressif/esp-idf-config/sdkconfig-esp32c6.defaults:55]:
```suggestion
# CONFIG_ESP_SYSTEM_HW_STACK_GUARD is not set
```

File: ports/espressif/esp-idf-config/sdkconfig-esp32c3.defaults [hunk 3]
```diff
@@ -46,4 +46,7 @@ CONFIG_NEWLIB_NANO_FORMAT=y
 
 # end of Component config
 
+# Workaround for https://github.com/espressif/esp-idf/issues/14456
```
Comment (tannewt) [inline ports/espressif/esp-idf-config/sdkconfig-esp32c3.defaults:49]: This comment will disappear when these files are automatically updated using `make update-all-sdkconfig`

**Review by tannewt — CHANGES_REQUESTED** [review]
Please match the generated style for disable so that when it gets auto-changed it is consistent. Yes, kconfig is weird using comments in a disable.

File: ports/espressif/esp-idf-config/sdkconfig-esp32c3.defaults (hunk 3 above)
Comment (dhalbert) [reply ports/espressif/esp-idf-config/sdkconfig-esp32c3.defaults]: I removed it. Reinforcing my dislike of Kconfig.

**Comment by dhalbert** [comment]
`CONFIG_ESP_SYSTEM_HW_STACK_GUARD` is enabled when `SOC_ASSIST_DEBUG_SUPPORTED` is defined, which is true for ESP32-C2, C3, C6, P4, and will eventually be true for C5 and C61, according to comments in ESP-IDF`. So I just turned it off with `not set` in sdkconfig.defaults.`

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**